test/itkDICOMOrientImageFilterGTest.cxx

```diff
@@ -0,0 +1,325 @@
+#include "itkRandomImageSource.h"
+#include "itkImage.h"
+#include <sstream>
+
+namespace
+{
+
+template <typename TPixelType, unsigned int D>
+void
+CheckImage(const itk::Image<TPixelType, D> * inputImage, const itk::Image<TPixelType, D> * resultsImage)
+{
+  using ImageType = itk::Image<TPixelType, D>;
+
+  itk::ImageRegionConstIteratorWithIndex<ImageType> iter(resultsImage, resultsImage->GetBufferedRegion());
+
+  while (!iter.IsAtEnd())
+  {
+    auto resultsIdx = iter.GetIndex();
+    auto point = resultsImage->template TransformIndexToPhysicalPoint<double>(resultsIdx);
+    auto inputIdx = inputImage->TransformPhysicalPointToIndex(point);
+
+    EXPECT_TRUE(inputImage->GetBufferedRegion().IsInside(inputIdx));
+
+    auto resultValue = iter.Value();
+    auto inputValue = inputImage->GetPixel(inputIdx);
+
+    EXPECT_EQ(resultValue, inputValue) << " result idx: " << resultsIdx << " input index: " << inputIdx;
+
+    ++iter;
+  }
+}
+
+} // namespace
+
+
+TEST(DICOMOrientImageFilter, NoOp)
+{
+  using ImageType = itk::Image<float, 3>;
+  auto filter = itk::DICOMOrientImageFilter<ImageType>::New();
+}
+
+
+TEST(DICOMOrientImageFilter, Streaming)
+{
+  using ImageType = itk::Image<uint32_t, 3>;
+
+  auto sourceFilter = itk::RandomImageSource<ImageType>::New();
+  sourceFilter->SetMin(-1000);
+  sourceFilter->SetMax(1000);
+  sourceFilter->SetSize({ 5, 6, 7 });
+  sourceFilter->UpdateLargestPossibleRegion();
+
+  ImageType::DirectionType direction;
+  direction.Fill(0.0);
+  direction(0, 0) = -1.0;
+  direction(1, 1) = -1.0;
+  direction(2, 2) = 1.0;
+  sourceFilter->SetDirection(direction);
+
+
+  using OrientFilterType = itk::DICOMOrientImageFilter<ImageType>;
+  using PermuiteArrayType = OrientFilterType::PermuteOrderArrayType;
+  using FlipArrayType = OrientFilterType::FlipAxesArrayType;
+  auto filter = OrientFilterType::New();
+  filter->SetInput(sourceFilter->GetOutput());
+
+  filter->GetOutput()->UpdateOutputInformation();
+  filter->GetOutput()->SetRequestedRegion(itk::Size<3>({ { 1, 2, 3 } }));
+  filter->Update();
+
+
+  CheckImage(sourceFilter->GetOutput(), filter->GetOutput());
+  EXPECT_EQ(filter->GetOutput()->GetBufferedRegion().GetSize(), itk::Size<3>({ { 1, 2, 3 } }));
+  EXPECT_EQ(filter->GetInputCoordinateOrientation(), OrientFilterType::OrientationEnum::RAS);
+  EXPECT_EQ(filter->GetPermuteOrder(), PermuiteArrayType({ { 0, 1, 2 } }));
+  EXPECT_EQ(filter->GetFlipAxes(), FlipArrayType({ { 1, 1, 0 } }));
+  {
+    ImageType::DirectionType d;
+    d.SetIdentity();
+    EXPECT_EQ(filter->GetOutput()->GetDirection(), d);
+  }
+}
+
+TEST(DICOMOrientImageFilter, Values)
+{
+  using ImageType = itk::Image<uint32_t, 3>;
+
+  auto sourceFilter = itk::RandomImageSource<ImageType>::New();
+  sourceFilter->SetMin(-1000);
+  sourceFilter->SetMax(1000);
+  sourceFilter->SetSize({ 5, 6, 7 });
+
+
+  using OrientFilterType = itk::DICOMOrientImageFilter<ImageType>;
+  using PermuiteArrayType = OrientFilterType::PermuteOrderArrayType;
```

Comment:
remove the i (PermuiteArrayType -> PermuteArrayType), but can be ignored as the error is consistent in the code, so no real harm.

Reply:
If it's invalid then the filter with just not perform any flip or permutations. We can refine this behavior later.